### ISSUE TITLE
Can't connect through corporate proxy

### ISSUE THREAD

**Issue by ccolic** [Cisco-AMP/amp4e_splunk_events_input#7]
Hi

I am trying to use this app to collect log data. But my Splunk host is behind a corporate proxy.

The proxy is a Cisco WSA, and in the logs I can see that the connection was allowed, but I guess that this somehow breaks the connection.

Is there any way to use this app behind a proxy? I tried to find some settings for the python pika library, but I didn't find anything.

I tried enabling the debug logs, but I didn't get any meaningful information from them:
```
[root@dev1 ~]# tailf /opt/splunk/var/log/splunk/amp4e_events_input.log
2018-09-17 13:04:50,463 DEBUG Amp4eEvents - Connecting to https://xxxx:xxxx@export-streaming.amp.cisco.com:443
2018-09-17 13:04:50,465 WARNING Amp4eEvents - Connection error (1537182290.47, <class 'pika.exceptions.ConnectionClosed'>: The AMQP connection was closed: ())! Reconnecting in about 3 seconds

[root@dev1 ~]# tailf /opt/splunk/var/log/splunk/splunkd.log
09-17-2018 12:51:01.656 +0200 ERROR ExecProcessor - message from "python /opt/splunk/etc/apps/amp4e_events_input/bin/amp4e_events_input.py" WARNING:pika.connection:Could not connect, 0 attempts left
09-17-2018 12:51:06.911 +0200 ERROR ExecProcessor - message from "python /opt/splunk/etc/apps/amp4e_events_input/bin/amp4e_events_input.py" WARNING:pika.adapters.base_connection:Connection to 52.70.56.136:443 failed: ('_ssl.c:660: Th
e handshake operation timed out',)
```

**Comment by samsonnguyen** [comment]
@ccolic 
Unfortunately, our app does not currently work behind a proxy. You may be able to configure a system wide proxy on the OS level (depending on operating system), in which case the proxy is transparent to the Splunk application.

For application specific proxies we do not currently have any plans to implement support (mainly due to other commitments). However, accept pull requests if you happen to be able to implement such a feature.

**Comment by samsonnguyen** [comment]
The app currently supports an environment variable
`HTTP_PROXY` and `HTTPS_PROXY`, which you can use to proxy requests to through a proxy.

You can specify the environment variable in a few ways, but mainly depends on how you are starting up splunk.


**splunk-launch.conf**
##simply add into the file

**systemd**
Most systemd installations accept custom overrides for a service
```
# Create /etc/systemd/system/splunkd.service.d/10-env.conf
# Assuming your systemd service unit is splunkd
HTTPS_PROXY=[proxy_url]
```

**Comment by arock1972** [comment]
Hi,
I have proxy configured on my system. I am trying to use TA / 1.1.4; stream creation going through, no issues. but no data at idx.. getting below error while investigating $SPLUNK_HOME/var/log/splunk/amp4*.log

2019-06-21 07:57:18,344 WARNING Amp4eEvents - Connection error (1561118238.34, <class 'pika.exceptions.ConnectionClosed'>: The AMQP connection was closed: ('Connection to 52.70.56.136:443 failed: timeout',))! Reconnecting in about 3 seconds
2019-06-21 07:57:29,360 WARNING Amp4eEvents - Connection error (1561118249.36, <class 'pika.exceptions.ConnectionClosed'>: The AMQP connection was closed: ('Connection to 52.70.56.136:443 failed: timeout',))! Reconnecting in about 3 seconds
2019-06-21 07:57:42,383 WARNING Amp4eEvents - Connection error (1561118262.38, <class 'pika.exceptions.ConnectionClosed'>: The AMQP connection was closed: ('Connection to 52.70.56.136:443 failed: timeout',))! Reconnecting in about 3 seconds
2019-06-21 07:57:57,405 WARNING Amp4eEvents - Connection error (1561118277.41, <class 'pika.exceptions.ConnectionClosed'>: The AMQP connection was closed: ('Connection to 52.70.56.136:443 failed: timeout',))! Reconnecting in about 3 seconds
